trailmap/commodity_dictionary.py

```diff
@@ -40,21 +46,11 @@ def get_commod_names(metadata, uitype, agent):
     return aliases
 
 
-def commodity_aliases(data, aliases):
-    '''Search for addional commodity aliases.'''
-    if isinstance(data, cyclus.jsoncpp.Value):
-        aliases.extend(data)
-    elif isinstance(data, str):
-        aliases.append(data)
-
-    return aliases
-
-
-def build_facility_dictionary(metadata):
+def build_facility_dictionary(metadata, archetypes):
     '''Identify commodities for each available archetype'''
     archetype_commods = {}
 
-    for archetype in metadata["specs"]:
+    for archetype in archetypes:
```

Comment:
I see that `archetypes = metadata["specs"]` is applied in line 20 but is there a reason for separating this from the metadata before calling `build_facility_dictionary`?

Reply:
Yes! Although `build_commod_dictionary` just turns the metadata specs into archetypes and passes it into `build_facility_dictionary`, one can imagine a use case where a user calls `build_facility_dictionary` on their own. They may want to pass a subset of archetypes and not the full list from the metadata, even though this is not the default behavior. This small change allows that